src/dao/api.dao.ts

```diff
@@ -38,7 +54,45 @@ export class ClaimDao {
     return createdClaim;
   };
 
-  createImages = async (claimId: number, userId: any, images: any[] = []) => {
+  async createClaimV2(userId: number, claim: CreateClaimV2Dto) {
+    const createdClaim = await prisma.claim.create({
+      data: {
+        issuerId: `${process.env.BASE_URL}/users/${userId}`,
+        issuerIdType: "URL",
+        subject: claim.subject,
+        amt: claim.amt,
+        claim: claim.claim,
+        object: claim.object,
+        statement: claim.statement,
+        aspect: claim.aspect,
+        howKnown: claim.howKnown,
+        sourceURI: claim.sourceURI,
+        effectiveDate: claim.effectiveDate,
+        confidence: claim.confidence,
+        stars: claim.stars,
+      },
+    });
+
+    return createdClaim;
+  }
+
+  async createImagesV2(claimId: number, userId: number, images: ImageDto[]): Promise<Image[]> {
+    if (!images.length) return [];
+
+    return prisma.$transaction(
+      images.map((x) =>
+        prisma.image.create({
+          data: {
+            ...x,
+            claimId: claimId,
+            owner: `${process.env.BASE_URL}/users/${userId}`,
```

Comment:
what if they are not our user?  there is a big assumption here that we have all the data, we might not

its more important that images are urls

and they MIGHT be anywhere

it may HAPPEN that a user puts them in our storage, but the whole idea is to support claims including those we do not create

Reply:
Humm...

Does that mean anyone can create a claim? The endpoint should be public without any protection/auth?

If so, should the user who creates the claim provide this owner field?

Reply:
we may import claims from many sources, and the conceptual owner of the image is the original external owner

this is fine for now i'm not acutally sure why we need this owner id i guess for people to delete their iamges later, so its a convenience for app users

That's fine for now.  we really don't know who will be the main users or if this will even be the usage pattern so just whatever is simplest is fine for now.  Mainly this is sitll just used for demos, we have to figure out what the real world use pattern will be.  A lot might be from spiders or ai agents inserting the data.

Reply:
I think with importing the claims the user will use some sort of a key that indicates who they are, this should work.

Or we could remove the field entirely. As you said we might not need to use it, if this is on a chain it should be immutable. And it's connected to a claim anyway (which makes it posible to get the owner from, in case we need to get this information)